process/asyncExecution/queue/blocksQueue_test.go

```diff
@@ -0,0 +1,186 @@
+package queue
+
+import (
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewHeadersQueue(t *testing.T) {
+	t.Parallel()
+
+	hq, err := NewBlocksQueue()
+	require.Nil(t, err)
+	require.NotNil(t, hq)
+	require.NotNil(t, hq.headerBodyPairs)
+	require.Equal(t, 0, len(hq.headerBodyPairs))
+	require.False(t, hq.IsInterfaceNil())
+}
+
+func TestHeadersQueue_Add(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil header should return error", func(t *testing.T) {
+		t.Parallel()
+		hq, _ := NewBlocksQueue()
+		err := hq.AddOrReplace(HeaderBodyPair{})
+		assert.Equal(t, common.ErrNilHeaderHandler, err)
+	})
+
+	t.Run("nil body should return error", func(t *testing.T) {
+		t.Parallel()
+		hq, _ := NewBlocksQueue()
+		err := hq.AddOrReplace(HeaderBodyPair{Header: &block.Header{Nonce: 1}})
+		assert.Equal(t, data.ErrNilBlockBody, err)
+	})
+
+	t.Run("valid header should be added", func(t *testing.T) {
+		t.Parallel()
```

Comment:
check the "replace" functionality as well

Reply:
added